src/node_process_methods.cc

```diff
@@ -471,6 +473,125 @@ static void ReallyExit(const FunctionCallbackInfo<Value>& args) {
   env->Exit(code);
 }
 
+inline char** copy_js_strings_array(Environment* env,
+                                    Local<Array> js_array,
+                                    int* target_length) {
+  Local<Context> context = env->context();
+  char** target = nullptr;
+  int length = js_array->Length();
+
+  CHECK_LT(length, INT_MAX);
+
+  target = new char*[length + 1];
+  target[length] = nullptr;
+
+  if (length > 0) {
+    for (int i = 0; i < length; i++) {
+      node::Utf8Value str(env->isolate(),
+                          js_array->Get(context, i).ToLocalChecked());
+      target[i] = strdup(*str);
+      CHECK_NOT_NULL(target[i]);
+    }
+  }
+
+  if (target_length) {
+    *target_length = length;
+  }
+
+  return target;
+}
+
+#ifdef __POSIX__
+inline int persist_standard_stream(int fd) {
+  int flags = fcntl(fd, F_GETFD, 0);
+
+  if (flags < 0) {
+    return flags;
+  }
+
+  flags &= ~FD_CLOEXEC;
+  return fcntl(fd, F_SETFD, flags);
+}
+
+static void Execve(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  Local<Context> context = env->context();
+
+  THROW_IF_INSUFFICIENT_PERMISSIONS(
+      env, permission::PermissionScope::kChildProcess, "");
+
+  CHECK_GT(args.Length(), 1);
+  CHECK(args[0]->IsString());
+
+  // Prepare arguments and environment
+  char** argv = nullptr;
+  char** envp = nullptr;
+  int envp_length = 0;
+
+  // Prepare arguments - Note that the pathname is always the first argument
+  if (args.Length() > 1) {
+    CHECK(args[1]->IsArray());
+
+    Local<Array> argv_array = args[1].As<Array>();
+
+    Local<Array> full_argv_array =
+        Array::New(env->isolate(), argv_array->Length() + 1);
+    full_argv_array->Set(context, 0, args[0].As<String>()).Check();
+    for (unsigned int i = 0; i < argv_array->Length(); i++) {
+      full_argv_array
+          ->Set(context, i + 1, argv_array->Get(context, i).ToLocalChecked())
+          .Check();
```

Comment:
If you don't want this to throw, you can surround this with USE() and remove .Check()

Reply:
?

Ideally, this should use neither of the two and just do proper error handling

Reply:
Indeed. this should probably be something like...

```c++
  if (full_argv_array->Set(...).IsNothing()) {
    return;
  }
```

Likewise the `Check()` and `ToLocalChecked()` uses immediately above should be changed to properly handle the error cases without resorting to abort.

This API call should only return nothing if an error is scheduled and returning immediately in that case allows v8 to proceed with throwing that error.

Reply:
I totally removed this function, no need for this anymore.